internal/handlers/commonparams/params.go

```diff
@@ -70,17 +70,19 @@ func GetWholeNumberParam(value any) (int64, error) {
 	}
 }
 
-// getWholeParamStrict validates the given value for find and count commands.
+// GetValidatedNumberParamWithBoundary converts and validates a value into a int64 number.
 //
-// If the value is valid, it returns its int64 representation,
-// otherwise it returns a command error with the given command being mentioned.
-func getWholeParamStrict(command string, param string, value any) (int64, error) {
+// The function checks the type, ensures it can be represented as a whole number,
+// isn't negative and falls within a given boundary and the limit of a 32-bit integer.
+//
+// It returns the processed integer value, or an error if the value fails validation.
+func GetValidatedNumberParamWithBoundary(command string, param string, value any, boundary int64) (int64, error) {
```

Comment:
It's a bit confusing that the boundary is int64, but the limit is int32... Is it on purpose?

Reply:
No, it wasn't made on purpose, changed the `minValue` type to `int32`.